File: platform/backend/AiDevRequest.API/Entities/PlatformUpgrade.cs
using System.ComponentModel.DataAnnotations;

namespace AiDevRequest.API.Entities;

public class PlatformUpgrade
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(100)]
    public required string UserId { get; set; }

    [MaxLength(50)]
    public string CurrentDotNetVersion { get; set; } = "net10.0";

    [MaxLength(50)]
    public string CurrentEfCoreVersion { get; set; } = "10.0";

    [MaxLength(50)]
    public string CurrentCSharpVersion { get; set; } = "14.0";

    public bool VectorSearchEnabled { get; set; } = true;
    public bool NativeJsonColumnsEnabled { get; set; } = true;
    public bool LeftJoinLinqEnabled { get; set; } = true;
    public bool PerformanceProfilingEnabled { get; set; } = false;

    public double AvgQueryTimeMs { get; set; } = 0;
    public double P95QueryTimeMs { get; set; } = 0;
    public double P99QueryTimeMs { get; set; } = 0;
    public long TotalQueriesExecuted { get; set; } = 0;
    public double CacheHitRate { get; set; } = 0;
    public double MemoryUsageMb { get; set; } = 0;
    public double CpuUsagePercent { get; set; } = 0;
    public double ThroughputRequestsPerSec { get; set; } = 0;

    public int VectorIndexCount { get; set; } = 0;
    public int VectorDimensions { get; set; } = 1536;
    public double VectorSearchAvgMs { get; set; } = 0;

    [MaxLength(20)]
    public string UpgradeStatus { get; set; } = "current";

    public string? FeatureFlagsJson { get; set; }
    public string? PerformanceHistoryJson { get; set; }
    public string? MigrationLogJson { get; set; }

    public DateTime? LastProfiledAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: platform/frontend/src/api/platformupgrade.ts
import { authFetch } from './auth'

const API = import.meta.env.VITE_API_URL || ''

export interface PlatformStatus {
  currentDotNetVersion: string
  currentEfCoreVersion: string
  currentCSharpVersion: string
  vectorSearchEnabled: boolean
  nativeJsonColumnsEnabled: boolean
  leftJoinLinqEnabled: boolean
  performanceProfilingEnabled: boolean
  upgradeStatus: string
}

export interface PerformanceMetrics {
  avgQueryTimeMs: number
  p95QueryTimeMs: number
  p99QueryTimeMs: number
  totalQueriesExecuted: number
  cacheHitRate: number
  memoryUsageMb: number
  cpuUsagePercent: number
  throughputRequestsPerSec: number
  lastProfiledAt: string | null
}

export interface VectorSearchStats {
  vectorIndexCount: number
  vectorDimensions: number
  vectorSearchAvgMs: number
  enabled: boolean
}

export interface DotNetFeature {
  name: string
  category: string
  description: string
  status: string
}

export interface BenchmarkResults {
  queryBenchmark: { net9Ms: number; net10Ms: number; improvement: string }
  serializationBenchmark: { net9Ms: number; net10Ms: number; improvement: string }
  startupBenchmark: { net9Ms: number; net10Ms: number; improvement: string }
  memoryBenchmark: { net9Mb: number; net10Mb: number; improvement: string }
  vectorSearchBenchmark: { avgMs: number; p95Ms: number; throughput: string }
}

export async function getUpgradeStatus(): Promise<PlatformStatus> {
  const res = await authFetch(`${API}/api/platform-upgrade/status`)
  if (!res.ok) throw new Error('Failed to load platform status')
  return res.json()
}

export async function updateSettings(data: {
  vectorSearchEnabled?: boolean
  nativeJsonColumnsEnabled?: boolean
  leftJoinLinqEnabled?: boolean
  performanceProfilingEnabled?: boolean
}): Promise<{ success: boolean }> {
  const res = await authFetch(`${API}/api/platform-upgrade/settings`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(data),
  })
  if (!res.ok) throw new Error('Failed to update settings')
  return res.json()
}

export async function getPerformanceMetrics(): Promise<PerformanceMetrics> {
  const res = await authFetch(`${API}/api/platform-upgrade/performance`)
  if (!res.ok) throw new Error('Failed to load performance metrics')
  return res.json()
}

export async function getVectorSearchStats(): Promise<VectorSearchStats> {
  const res = await authFetch(`${API}/api/platform-upgrade/vector-search`)
  if (!res.ok) throw new Error('Failed to load vector search stats')
  return res.json()
}

export async function getDotNetFeatures(): Promise<DotNetFeature[]> {
  const res = await authFetch(`${API}/api/platform-upgrade/features`)
  if (!res.ok) throw new Error('Failed to load features')
  return res.json()
}

export async function runBenchmark(): Promise<BenchmarkResults> {
  const res = await authFetch(`${API}/api/platform-upgrade/run-benchmark`, { method: 'POST' })
  if (!res.ok) throw new Error('Failed to run benchmark')
  return res.json()
}
